codecparsers/mpeg2_parser.cpp

```diff
@@ -173,7 +173,7 @@ namespace MPEG2 {
 
     SeqExtension::SeqExtension() { memset(this, 0, sizeof(*this)); }
 
-    SeqHeader::SeqHeader() { memset(this, 0, sizeof(*this)); }
+    SeqHeader::SeqHeader() { memset((void*)this, 0, sizeof(*this)); }
```

Comment:
if we need this, why we do not need to add the same thing for SeqExtension?

Reply:
Thanks for your review. GCC9 feels uneasy with non-trival classes.

Reply:
but SeqExtension has constructor like SeqHeader, right?
thanks

Reply:
I believe definition of  Trival class includes not only constructors but also operators, destructors, data members, functions, baseclass. All these should be trival  in order to define a class trival.

useful link:
https://stackoverflow.com/questions/30096911/which-rules-determine-whether-an-object-is-trivially-copyable

Reply:
From a quick view, I can say SeqExtension has no user defined members, but this is not the case with SeqHeader.

Reply:
```
    struct SeqHeader {
        SeqHeader();

        uint32_t horizontal_size_value;
        uint32_t vertical_size_value;
        uint32_t aspect_ratio_info;
        uint32_t frame_rate_code;
        uint32_t bit_rate_value;
        bool marker_bit;
        uint32_t vbv_buffer_size_value;
        bool constrained_params_flag;
        QuantMatrices quantizationMatrices;
    };
```

You are right. SeqHeader has the quantizationMatrices, the memset will override the value set in QuantMatrices().
Could you help change it to 

`SeqHeader::SeqHeader() { memset((void*)this, 0, offsetof(SeqHeader, quantizationMatrices)); }
`

thanks

Reply:
Done